### ISSUE TITLE
stdout/err can still be redirected to system.log

### ISSUE THREAD

**Issue by szhu** [mkalmes/brewupdate#5]
From [`man launchd.plist`](https://developer.apple.com/library/mac/documentation/Darwin/Reference/Manpages/man5/launchd.plist.5.html):

> **StandardOutPath &lt;string>**
> This optional key specifies what file should be used for data being sent to stdout when using stdio(3).
> 
> **StandardErrorPath &lt;string>**
> This optional key specifies what file should be used for data being sent to stderr when using stdio(3).

**Comment by mkalmes** [comment]
Logging to system.log requires root permissions. OS X Administrator privileges are not sufficient.
Logging to files under `~/Library/Logs` will not work due to the lack of shell expansion.

A workaround would be to use make/rake to setup the logging path for the current user.

**Comment by szhu** [comment]
What if `brewupdate` is made into a LaunchDaemon that logs to `system.log` but runs Homebrew as the admin user (via the `UserName` key)? This makes sense because since there's usually one installation of Homebrew anyway (the one in `/usr/local`).

make/rake sounds like a good approach for doing this for per-user installations.

**Comment by mkalmes** [comment]
According to the [homebrew wiki](https://github.com/Homebrew/homebrew/wiki/FAQ#wiki-sudo), it is not necessary to install or even run `brew` with `sudo`.

It seems that a deployment script is the only way to set `StandardOutPath`.

**Comment by szhu** [comment]
> According to the homebrew wiki, it is not necessary to install or even run `brew` with `sudo`.
1. We're only running the launchd job as root so we can log, then we're dropping down to the Homebrew user for running Homebrew. Homebrew shouldn't be run as root because it is huge; a script that simply runs Homebrew as the Homebrew user and logs to the system log is not very dangerous.
2. This modification is intended for the majority of users who use Homebrew's default prefix of `/usr/local`, which sudo was initially required for `chown`. Users who choose override the default for the prefix can also override the default for this launchd job.

> It seems that a deployment script is the only way to set StandardOutPath.

What do you mean by this?

**Comment by mkalmes** [comment]
> > It seems that a deployment script is the only way to set StandardOutPath.
> 
> What do you mean by this?

Using make or rake for shell expansion which is missing in `launchd`.

**Comment by szhu** [comment]
Actually after reading [this](http://apple.stackexchange.com/questions/83827/where-does-homebrew-log) it looks like `~/Library/Logs/Homebrew/` is the best place to store a log file. It's probably worth having a deployment script or at least a few lines to show people how to add in `StandardOutPath` themselves, something along the lines of:

```
mkdir -p ~/Library/Logs/Homebrew/brewupdate
defaults write ~/Library/LaunchAgents/net.mkalmes.brewupdate StandardOutPath ~/Library/Logs/Homebrew/brewupdate/brewupdate.log
```

**Comment by mkalmes** [comment]
Will be addressed w/ #10.
